bfvmm/tests/hve/arch/intel_x64/ept/ept_test_support.h

```diff
@@ -0,0 +1,296 @@
+//
+// Bareflank Extended APIs
+//
+// Copyright (C) 2015 Assured Information Security, Inc.
+//
+// This library is free software; you can redistribute it and/or
+// modify it under the terms of the GNU Lesser General Public
+// License as published by the Free Software Foundation; either
+// version 2.1 of the License, or (at your option) any later version.
+//
+// This library is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+// Lesser General Public License for more details.
+//
+// You should have received a copy of the GNU Lesser General Public
+// License along with this library; if not, write to the Free Software
+// Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA
+
+#include <catch/catch.hpp>
+#include <hippomocks.h>
+
+#include "hve/arch/intel_x64/ept.h"
+#include <bfvmm/memory_manager/memory_manager.h>
+
+namespace eapis
+{
+namespace intel_x64
+{
+namespace ept
+{
+
+constexpr const uintptr_t mock_pml4_hpa = 0x000ABCD000000000ULL;
+constexpr const uintptr_t mock_pdpt_hpa = 0x0000000ABCD00000ULL;
+constexpr const uintptr_t mock_pd_hpa = 0x0000000123400000ULL;
+constexpr const uintptr_t mock_pt_hpa = 0x0000000DCBA00000ULL;
+constexpr const uintptr_t mock_page_hpa = 0x000000000F00D000ULL;
+
+constexpr const uint64_t mock_pml4e_offset = 0xFF8;
+constexpr const uint64_t mock_pdpte_offset = 0xFF8;
+constexpr const uint64_t mock_pde_offset = 0xFF8;
+constexpr const uint64_t mock_pte_offset = 0xFF8;
+constexpr const uint64_t mock_page_offset = 0xA55A;
+
+constexpr const uintptr_t mock_1g_hpa = 0xFFFFC0000000ULL;
+constexpr const uintptr_t mock_2m_hpa = 0xFFFFFFE00000ULL;
+constexpr const uintptr_t mock_4k_hpa = 0xFFFFFFFFF000ULL;
+
+// The following gpa is mapped into all mocked memory maps
+constexpr const uintptr_t g_mapped_gpa = 0x0000FFFFFFFFF000ULL;
+
+// The following gpa is not mapped by any entries in any mocked memory maps
+constexpr const uintptr_t g_unmapped_gpa = 0x0000000000000000ULL;
+
+std::map<void *, uintptr_t> g_mock_mem;
+
+static volatile uintptr_t g_next_phys_addr = 0x00000000F00D0000;
+uintptr_t
+mock_virtptr_to_physint(void * gva)
+{
+    if(g_mock_mem.count(gva)){
```

Comment:
Need spaces between control flow construct and the opening brace. Did format not catch this? If so I can tweak it

Reply:
I didn't see this the first time around; my bad

Reply:
format seems to have missed numerous things like this. Either I'm not using right or something is broken. I'll look into it first thing tomorrow